ibis/pyspark/compiler.py

```diff
@@ -1489,9 +1489,30 @@ def compile_not_null(t, expr, scope, **kwargs):
 
 
 @compiles(ops.ElementWiseVectorizedUDF)
-def compile_elementwise_udf(t, expr, scope):
+def compile_elementwise_udf(t, expr, scope, **kwargs):
     op = expr.op()
     spark_output_type = spark_dtype(op._output_type)
     spark_udf = pandas_udf(op.func, spark_output_type, PandasUDFType.SCALAR)
     func_args = (t.translate(arg, scope) for arg in op.func_args)
     return spark_udf(*func_args)
+
+
+@compiles(ops.ReductionVectorizedUDF)
+def compile_reduction_udf(t, expr, scope, context=None, **kwargs):
+    op = expr.op()
+
+    spark_output_type = spark_dtype(op._output_type)
+    spark_udf = pandas_udf(
+        op.func, spark_output_type, PandasUDFType.GROUPED_AGG
+    )
+    func_args = (t.translate(arg, scope) for arg in op.func_args)
+
+    col = spark_udf(*func_args)
+    if context == AggregationContext.ENTIRE:
```

Comment:
Technically could make this if-else block into a helper function that can be reused, but we already have a couple aggregation-related compilation or helper functions. They're getting a little hard to keep track of so I think a bit of code repetition here is preferable

- [`compile_aggregation`](https://github.com/ibis-project/ibis/blob/5893bbc259477a63f4182540ed6d1462832935fa/ibis/pyspark/compiler.py#L289)
- [`_compile_agg`](https://github.com/ibis-project/ibis/blob/master/ibis/pyspark/compiler.py#L281)
- [`compile_aggregator`](https://github.com/ibis-project/ibis/blob/master/ibis/pyspark/compiler.py#L326)